anchore_engine/services/policy_engine/engine/feeds/client.py

```diff
@@ -247,7 +423,87 @@ def _extract_response_data(self, response_text):
         return next_token, response_text, count
 
 
-def get_client(
+class GrypeDBServiceClient(IFeedSource):
+    RETRY_COUNT = 3
+
+    def __init__(
+        self,
+        grype_db_endpoint: str,
+        http_client: HTTPBasicAuthClient,
+    ):
+        self.feed_url = grype_db_endpoint
+        self.http_client = http_client
+
+    def list_feeds(self) -> FeedList:
+        return FeedList(
+            feeds=[
+                FeedAPIRecord(
+                    name="grypedb",
+                    description="grypedb feed",
+                    access_tier="0",
+                )
+            ]
+        )
+
+    def list_feed_groups(self, feed: str) -> FeedGroupList:
+        return FeedGroupList(
+            groups=[
+                FeedAPIGroupRecord(
+                    name="grypedb:vulnerabilities",
+                    description="grypedb:vulnerabilities group",
+                    access_tier="0",
+                )
+            ]
+        )
+
+    def _get_raw_feed_group_data(
+        self,
+    ) -> Tuple[Dict, HTTPClientResponse]:
+        logger.info("Downloading grypedb listing.json from {}".format(self.feed_url))
+        listing_response = self.http_client.execute_request(
+            requests.get, self.feed_url, retries=self.RETRY_COUNT
+        )
+        if not listing_response.success:
+            raise HTTPStatusException(listing_response)
+        listings_json = json.loads(listing_response.content.decode("utf-8"))
+        db_listing = listings_json.get("available").get("2")[0]
```

Comment:
@wagoodman @alfredodeza is this something that could be updated in grype?

Reply:
I'm sure I'm missing context here. But why should the schema version be configurable? Is the Grype version configurable?

Reply:
Not user configurable. The download logic should take the schema version as an input rather than hardcoding it. An enhancement would be asking grype for required db version so the download downloads the correct database

Reply:
The required schema version is exposed by the `grype version` command. Is the problem the output format?

```
$ grype version
Application:          grype
Version:              0.11.0
BuildDate:            2021-04-22T21:51:14Z
GitCommit:            0a46e54aae837135f73d2ff18a84061189ee6f28
GitTreeState:         clean
Platform:             darwin/amd64
GoVersion:            go1.16.3
Compiler:             gc
Supported DB Schema:  2
```

Reply:
thanks! we didn't know that. that's what we needed

Reply:
https://github.com/anchore/anchore-engine/issues/1036